features/karaf/openhab-addons/src/main/feature/feature.xml

```diff
@@ -861,10 +861,20 @@
 
     <feature name="openhab-binding-xmppclient" description="XMPP Client Binding" version="${project.version}">
         <feature>openhab-runtime-base</feature>
-        <bundle dependency="true">wrap:mvn:org.igniterealtime.smack/smack-java7/4.3.1$Bundle-Name=Smack%20XMPP%20Java7&amp;Bundle-SymbolicName=org.igniterealtime.smack.smack-java7&amp;Bundle-Version=4.3.1</bundle>
-        <bundle dependency="true">wrap:mvn:org.igniterealtime.smack/smack-extensions/4.3.1$Bundle-Name=Smack%20XMPP%20Extensions&amp;Bundle-SymbolicName=org.igniterealtime.smack.smack-extensions&amp;Bundle-Version=4.3.1</bundle>
-        <bundle dependency="true">wrap:mvn:org.igniterealtime.smack/smack-im/4.3.1$Bundle-Name=Smack%20XMPP%20IM&amp;Bundle-SymbolicName=org.igniterealtime.smack.smack-im&amp;Bundle-Version=4.3.1</bundle>
-        <bundle dependency="true">wrap:mvn:org.igniterealtime.smack/smack-tcp/4.3.1$Bundle-Name=Smack%20XMPP%20TCP&amp;Bundle-SymbolicName=org.igniterealtime.smack.smack-tcp&amp;Bundle-Version=4.3.1</bundle>
+        <!-- Main dependencies -->
+        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-java7/4.3.1</bundle>
```

Comment:
They are now doubled

Reply:
I'm trying to figure out the build error in Travis. Where I can find right budle name?

For example the MANIFEST.MF for smack-im:

```
Manifest-Version: 1.0
Bundle-SymbolicName: org.igniterealtime.smack.im
Implementation-Version: 4.3.1
Built-Date: 2018-10-14
Built-By: flo
Bundle-ManifestVersion: 2
Bnd-LastModified: 1539370816000
Import-Package: <.... skipped ....>
Built-Gradle: 4.10.2
Built-JDK: 1.8.0_172
Tool: Bnd-3.4.0.201707252008
Implementation-GitRevision: 4.3.1-4.3
Export-Package: <.... skipped ....>
Bundle-Version: 4.3.1
Bundle-Name: smack-im
Created-By: 1.8.0_172 (Oracle Corporation)
```

What is the correct entry for Karaf?

Maybe:
```
<bundle dependency="true">mvn:org.igniterealtime.smack.im/smack-im/4.3.1</bundle>
```
?

Reply:
What should I do if one of the dependencies does not support OSGI? Should I describe it separately with a wrapper?

Reply:
There was no build error in travis. The error is from the karaf feature verification. Karaf tried to resolve the mentioned bundle and didn't find it.

Yes: Either OSGi manifest is included or you need to "wrap:" it with the symbolic name and stuff like you had it.

But your jabber library seem to use org.jivesoftware.smack and not org.igniterealtime.smack. Maybe you can update the jabber library or similar?

Reply:
> But your jabber library seem to use org.jivesoftware.smack and not org.igniterealtime.smack. Maybe you can update the jabber library or similar?

After version 3.1.0, all smack artifacts was moved to *org.igniterealtime.smack* repository. But package names inside is not changed (*org.jivesoftware.smack.**).

Thanks for the help.

Reply:
Karaf verification is failing again. 

My feature description:
```
<feature name="openhab-binding-xmppclient" description="XMPP Client Binding" version="${project.version}">
        <feature>openhab-runtime-base</feature>
        <feature prerequisite="true">wrap</feature>
        <!-- Main dependencies -->
        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-java7/4.3.1</bundle>
        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-extensions/4.3.1</bundle>
        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-im/4.3.1</bundle>
        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-tcp/4.3.1</bundle>
        <!-- Dependencies of dependencies -->
        <bundle dependency="true">wrap:mvn:xpp3/xpp3/1.1.4c$Bundle-Name=XPP3%20Library&amp;Bundle-SymbolicName=xpp3-xpp3&amp;Bundle-Version=1.1.4</bundle>
        <bundle start-level="80">mvn:org.openhab.addons.bundles/org.openhab.binding.xmppclient/${project.version}</bundle>
    </feature>
```

Error log:
```
resolution:=mandatory [caused by: Unable to resolve org.openhab.binding.xmppclient/2.5.0.201903271304: missing requirement [org.openhab.binding.xmppclient/2.5.0.201903271304] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.jivesoftware.smack)(version>=4.3.0)(!(version>=5.0.0)))"]]
```

I have no ideas how to make it works...

Reply:
As I said. If you look at the manifest of your embedded smack:
```
Manifest-Version: 1.0
Bundle-SymbolicName: org.igniterealtime.smack.java7
Implementation-Version: 4.3.3
Built-Date: 2019-03-14
Built-By: flo
Bundle-ManifestVersion: 2
Bnd-LastModified: 1552598958000
Import-Package: javax.naming,javax.naming.ldap,javax.net.ssl,javax.sec
 urity.auth.kerberos,javax.security.auth.x500,org.jivesoftware.smack;v
 ersion="[4.3,5)",org.jivesoftware.smack.initializer;version="[4.3,5)"
 ,org.jivesoftware.smack.util;version="[4.3,5)",org.jivesoftware.smack
 .util.stringencoder;version="[4.3,5)",org.jivesoftware.smack.util.str
 ingencoder.java7;version="[4.3,5)",org.minidns.util;version="[0.3,1)"
Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.7))"
Built-Gradle: 4.10.2
Built-JDK: 1.8.0_192
Tool: Bnd-3.4.0.201707252008
Implementation-GitRevision: 4.3.3-4.3
Export-Package: org.jivesoftware.smack.java7;version="4.3.3";uses:="ja
 vax.net.ssl,org.jivesoftware.smack.initializer",org.jivesoftware.smac
 k.util.stringencoder.java7;version="4.3.3";uses:="org.jivesoftware.sm
 ack.util.stringencoder"
Bundle-Version: 4.3.3
Bundle-Name: smack-java7
Created-By: 1.8.0_192 (Oracle Corporation)
```

Maybe it is the Java-7 requirement? `Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.7))"`

Reply:
@openhab/2-x-add-ons-maintainers I'm also out of ideas here.

Reply:
Last version was build successed. I added all dependencies to feature.xml

```
    <feature name="openhab-binding-xmppclient" description="XMPP Client Binding" version="${project.version}">
        <feature>openhab-runtime-base</feature>
        <feature prerequisite="true">wrap</feature>
        <!-- Main dependencies -->
        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-java7/4.3.1</bundle>
        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-extensions/4.3.1</bundle>
        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-im/4.3.1</bundle>
        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-tcp/4.3.1</bundle>
        <!-- Dependencies of dependencies -->
        <bundle dependency="true">mvn:org.jxmpp/jxmpp-core/0.6.3</bundle>
        <bundle dependency="true">mvn:org.jxmpp/jxmpp-jid/0.6.3</bundle>
        <bundle dependency="true">mvn:org.jxmpp/jxmpp-util-cache/0.6.3</bundle>
        <bundle dependency="true">mvn:org.minidns/minidns-core/0.3.3</bundle>
        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-core/4.3.1</bundle>
        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-resolver-javax/4.3.1</bundle>
        <bundle dependency="true">mvn:org.igniterealtime.smack/smack-sasl-javax/4.3.1</bundle>
        <bundle dependency="true">wrap:mvn:xpp3/xpp3/1.1.4c$Bundle-Name=XPP3%20Library&amp;Bundle-SymbolicName=xpp3-xpp3&amp;Bundle-Version=1.1.4</bundle>
        <bundle start-level="80">mvn:org.openhab.addons.bundles/org.openhab.binding.xmppclient/${project.version}</bundle>
    </feature>
```

But Jenkins looks strange.

Reply:
Jenkins does not need to agree, travis is important. Jenkins fails a lot, mostly because the p2 repos for the old buildsystem seem to be unstable.